src/main/java/games/strategy/engine/data/GameParser.java

```diff
@@ -76,67 +76,64 @@ private GameParser(final String mapName) {
    * @return A complete {@link GameData} instance that can be used to play the game.
    */
   public static GameData parse(final String mapName, final InputStream stream)
-      throws GameParseException, SAXException, EngineVersionException {
-    return parse(mapName, stream, new AtomicReference<>());
-  }
-
-  /**
-   * Performs a deep parse of the game definition contained in the specified stream.
-   *
-   * @param gameName Receives the game name parsed from the game definition.
-   *
-   * @return A complete {@link GameData} instance that can be used to play the game.
-   */
-  public static GameData parse(final String mapName, final InputStream stream, final AtomicReference<String> gameName)
-      throws GameParseException, SAXException, EngineVersionException {
+      throws GameParseException, EngineVersionException {
     checkNotNull(mapName);
     checkNotNull(stream);
-    checkNotNull(gameName);
 
-    return new GameParser(mapName).parse(stream, gameName);
+    return new GameParser(mapName).parse(stream);
   }
 
-  private GameData parse(final InputStream stream, final AtomicReference<String> gameName)
-      throws GameParseException, SAXException, EngineVersionException {
+  private GameData parse(final InputStream stream) throws GameParseException, EngineVersionException {
     final Element root = parseDom(stream);
-    parseMapProperties(root, gameName);
+    parseMapProperties(root);
     parseMapDetails(root);
     return data;
   }
 
+  private GameParseException newGameParseException(final String message) {
+    return newGameParseException(message, null);
+  }
+
+  private GameParseException newGameParseException(final String message, final @Nullable Throwable cause) {
```

Comment:
This method name seems a little bit odd, maybe something like "generateException" sounds better?
Also I'd suggest changing the behaviour so this method also throws this exception but still returns GameParseException so this method works similar to the guava Throwables.propagate method that has been deprecated.

Reply:
> This method name seems a little bit odd

Well, it's just a factory method, and our convention for factory methods seems to be `newXXX`, where `XXX` is the type being created.  The JDK uses a similar convention extensively (e.g. look at the JDK index entry for [N](https://docs.oracle.com/javase/8/docs/api/index-files/index-14.html)).

> Also I'd suggest changing the behaviour so this method also throws this exception but still returns GameParseException so this method works similar to the guava Throwables.propagate method that has been deprecated

One of the reasons Google gives for why they deprecated `Throwables#propagate()` is that the recommended usage

```java
throw Throwables.propagate(e);
```

is an attempt to outsmart the compiler's reachability detector.  They cited that many people did not follow the recommendation (i.e. they didn't throw the return value), which lead to the introduction of dummy statements that were effectively unreachable.  It seems we could have the same issue in the future for someone new to the codebase.

You'll note that the new `Throwables#throwIfXXX()` methods return `void`, and thus require users to terminate a sequence of those calls with an explicit `throw`.  Guava doesn't give users the opportunity to make the same mistake as `propagate()`.  Similarly, there's only one way to use `newGameParseException()`: you must throw the return value.

So, I guess my question is, why would we want to imitate behavior Google specifically identified as problematic?

Reply:
Good Point.
Nevermind then.